docs/current/sdk/python/628797-get-started.md

```diff
@@ -0,0 +1,167 @@
+---
+slug: /sdk/python/628797/get-started
+---
+
+# Get Started with the Dagger Python SDK
+
+## Introduction
+
+This tutorial teaches you the basics of using Dagger in Python. You will learn how to:
+
+- Install the Python SDK
+- Create a Python CI tool to test an application
+- Improve the Python CI tool to test the application against multiple Python versions
+
+## Requirements
+
+This tutorial assumes that:
+
+- You have a basic understanding of the Python programming language. If not, [read the Python tutorial](https://www.python.org/about/gettingstarted/).
+- You have a Python development environment with Python 3.x and Poetry. If not, install [Python](https://www.python.org/downloads/) and [Poetry](https://python-poetry.org/docs/#installation).
+- You have Docker installed and running on the host system. If not, [install Docker](https://docs.docker.com/engine/install/).
+- You have a Python application with tests defined.
+
+:::note
+This tutorial creates a CI tool to test your Python application against multiple Python versions. If you don't have a Python application already, clone an existing Python project with a well-defined test suite before proceeding. A good example is the [FastAPI](https://github.com/tiangolo/fastapi) library, which you can clone as below:
+
+```shell
+git clone https://github.com/tiangolo/fastapi
+```
+
+:::
+
+## Step 1: Create a Dagger client in Python
+
+The first step is to create a Python package for the Python CI tool. Run the following commands in your project directory.
+
+```shell
+poetry new ci
+```
+
+Next, create a new file named `ci/test.py` and add the following code to it.
+
+```python file=snippets/get-started/step1/test.py
+```
+
+This Python stub imports the Dagger SDK and defines an asynchronous function named `test()`. This `test()` function performs the following operations:
+
+- It creates a Dagger client with `dagger.Connection()`. This client provides an interface for executing commands against the Dagger engine.
+- It uses the client's `container().from_()` method to initialize a new container from a base image. In this example, the base image is the `python:3.10-slim-buster` image. This method returns a `Container` representing an OCI-compatible container image.
+- It uses the `Container.exec()` method to define the command to be executed in the container - in this case, the command `python -V`, which returns the Python version string. The `exec()` method returns a revised `Container` with the results of command execution.
+- It retrieves the output stream of the last executed command with the `Container.stdout()` method and prints its contents.
+
+## Step 2: Add the Dagger Python SDK
+
+Install the Dagger Python SDK using `poetry` as shown below:
+
+```shell
+cd ci
+poetry install
+cd ..
```

Comment:
This switching back and forth between directories is inconvenient but I couldn't find a better way to do this, as `poetry install` can't be run from a different directory. Seems to be a poetry limitation and there is already multiple requests for this and a fix awaiting release https://github.com/python-poetry/poetry/pull/6810. Until then, we can go with this or maybe @helderco can suggest a better approach.

Reply:
I suggest don't use poetry, just add a notice like "It is recommended that you set up a [virtualenv](https://docs.python-guide.org/dev/virtualenvs/)...".

Reply:
Modified